### ISSUE TITLE
Fix DART booster predictions when rate_drop > 0

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#183]
When using xgboost's DART booster with `rate_drop > 0`, `tidypredict_fit()` produces incorrect predictions (diff ~0.11 from native `predict()`). DART normalizes tree outputs when dropout is used, but tidypredict doesn't account for this.

DART with `rate_drop = 0` works correctly and should have a test added.

## Reprex

```r
library(xgboost)
library(tidypredict)

set.seed(123)

xgb_data <- xgb.DMatrix(
  as.matrix(mtcars[, -9]),
  label = mtcars$am
)

model <- xgb.train(
  params = list(
    max_depth = 2L,
    objective = "reg:squarederror",
    base_score = 0.5,
    booster = "dart",
    rate_drop = 0.3,
    seed = 123
  ),
  data = xgb_data,
  nrounds = 4L,
  verbose = 0
)

# Compare predictions
fit <- tidypredict_fit(model)
tidy_preds <- rlang::eval_tidy(fit, mtcars)
native_preds <- predict(model, xgb_data)

head(data.frame(
  tidy = tidy_preds,
  native = native_preds,
  diff = tidy_preds - native_preds
))
#>        tidy    native       diff
#> 1 0.9485185 0.8348881  0.1136304
#> 2 0.9485185 0.8348881  0.1136304
#> 3 0.9485185 0.8348881  0.1136304
#> 4 0.0420046 0.1581042 -0.1160996
#> 5 0.0420046 0.1581042 -0.1160996
#> 6 0.0420046 0.1581042 -0.1160996
```
